tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/add-operation.bal

```diff
@@ -368,13 +368,73 @@ function testStringXmlSubtypesAddition() {
     string:Char c = "d";
     xml x1 = xml `efg`;
     xml:Text x2 = xml `text`;
+    string:Char|Strings b = "d";
+    Strings|Chars e = "bar";
+    bar d = "B";
 
     assertEquality(s + x2, xml `abctext`);
     assertEquality(x2 + s, xml `textabc`);
     assertEquality(c + x1, xml `defg`);
     assertEquality(x1 + c, xml `efgd`);
     assertEquality(c + x2, xml `dtext`);
     assertEquality(x2 + c, xml `textd`);
+    assertEquality(x2 + b, xml `textd`);
+    assertEquality(x2 + e, xml `textbar`);
+    assertEquality(x2 + d, xml `textB`);
+    assertEquality(x1 + b, xml `efgd`);
+    assertEquality(x1 + e, xml `efgbar`);
+    assertEquality(x1 + d, xml `efgB`);
+}
+
+type Chars "B"|"bar";
+type bar Chars|string;
+
+function testStringSubtypesAddition() {
+    string a = "abc";
+    string:Char|Strings b = "d";
+    Strings|Chars c = "bar";
+    bar d = "B";
+    string:Char e = "e";
+    assertEquality(a + c, "abcbar");
```

Comment:
We aren't testing the static types here. Better to have at least a few that are part of a variable declaration or assignment with the specific type.

Reply:
For XML, there may be [known issues](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aopen+xml+additive).

Reply:
Updated.